### PR TITLE
feat: update hg-slate-yjs with slate-yjs master branch

### PR DESCRIPTION
**Pull request by JarnDev**
Update hg-slate-yjs with master of slate-yjs and add hugo tests

resolves HGA-1192

### CODE REVIEW

File: src/applyToSlate/hugoMapEvent.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  // const newProperties = Object.fromEntries(
+  //   keyChanges.map(([key, info]) => [
+  //     key,
+  //     info.action === 'delete' ? null : targetSyncElement.get(key),
+  //   ])
+  // );
+
+  // const properties = Object.fromEntries(
+  //   keyChanges.map(([key]) => [key, targetElement[key]])
+  // );
```
Comment (chrisfsampaio) [inline src/applyToSlate/hugoMapEvent.ts:49]: Left over?

File: src/plugin/cursorEditor.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  e.onChange = () => {
+    setTimeout(() => CursorEditor.updateCursor(e), 0);
+
+    if (onChange) {
+      onChange();
+    }
+  };
+
+  return e;
+}
```
Comment (chrisfsampaio) [inline src/plugin/cursorEditor.ts:57]: I don't think we need this file, we're handling awareness in the frontend directly.

File: src/plugin/useCursors.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+      return ranges;
+    },
+    [cursors]
+  );
+
+  return { decorate, cursors };
+};
+
+export default useCursors;
```
Comment (chrisfsampaio) [inline src/plugin/useCursors.ts:93]: Same for this

File: test/apply/apply.test.ts [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+        path: [0, 2],
+        properties: { strong: true },
+        newProperties: {},
+        type: 'set_node',
+      },
+    ],
+    [createNode('paragraph', 'hotel uniform golf oscar')],
+  ],
+  [
+    // TODO: check the new way to add text with marks
```
Comment (chrisfsampaio) [inline test/apply/apply.test.ts:281]: Anything I need to test in the UI?

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,11845 @@
+{
```
Comment (chrisfsampaio) [inline package-lock.json:1]: Do we need this file? We already have a `yarn.lock`…

File: src/applyToSlate/hugoMapEvent.ts (hunk 1 above)
Comment (JarnDev) [reply src/applyToSlate/hugoMapEvent.ts]: Yeah, this file was removed

File: test/apply/apply.test.ts (hunk 4 above)
Comment (JarnDev) [reply test/apply/apply.test.ts:281]: The only thing would be confirm if the test steps is robust enough with the real usage situation.

**Review by chrisfsampaio — APPROVED** [review]

**Pull request merged**